.github/workflows/release.yml

```diff
@@ -0,0 +1,145 @@
+            checksums = {}
+            for (const r of releases["data"]) {
+              if (r.draft && `refs/tags/${r.tag_name}` == "${{ github.ref }}") {
+                for (const asset of r.assets) {
+                  var release_asset = await github.rest.repos.getReleaseAsset({ headers: {accept: `application/octet-stream`}, accept: `application/octet-stream`, owner: owner, repo: repo, asset_id: asset.id });
+                  const hash = crypto.createHash('sha256');
+                  let http_promise = new Promise((resolve, reject) => {
+                    https.get(release_asset.url, (stream) => {
+                      stream.on('data', function (data) {
+                        hash.update(data);
+                      });
+                      stream.on('end', function () {
+                        checksums[asset.name]= hash.digest('hex');
+                        resolve(`${asset.name}`);
+                      });
+                    });
+                  });
+                  await http_promise;
+                }
+              }
+            }
+            console.log(checksums)
+            return `${checksums['release.yml']}  ./tmp/release.yml
+            ${checksums['package.yml']}  ./tmp/package.yml
+            ${checksums['metadata.yml']}  ./tmp/metadata.yml`
+      - name: Verify uploaded artifacts
+        if: startsWith(github.ref, 'refs/tags/')
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: |
+          set -e -x
+          cat ./tmp/checksums.txt
+          diff ./tmp/checksums.txt <(cat <<EOF
+          ${{steps.get-checksums-from-draft-release.outputs.result}}
+          EOF
+          )
+      - name: Run PackageCR Validation Tests
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -e -x
+
+          # deploy admin sa for secretgen-controller - enough permissions for SGC to be deployed properly
+          kapp deploy -a rbac -f https://raw.githubusercontent.com/vmware-tanzu/carvel-kapp-controller/develop/examples/rbac/cluster-admin.yml -y
+
+          # Kapp-controller is needed for our PackageInstall
+          kapp deploy -a kc -f https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml -y
+
+          constraintVersion="${{ github.ref_name }}"
+          cat << EOF > pkgi.yml
+          ---
+          apiVersion: packaging.carvel.dev/v1alpha1
+          kind: PackageInstall
+          metadata:
+            name: secretgen-controller
+          spec:
+            serviceAccountName: cluster-admin-sa
+            packageRef:
+              refName: secretgen-controller.carvel.dev
+              apiVersion: packaging.carvel.dev/v1alpha1
+              versionSelection:
+                constraints: "${constraintVersion:1}"
+          EOF
+
+          # download the generated package and metadata from github draft release
+          gh release download "${{ github.ref_name }}" -p "*.yml"
```

Comment:
could we just use files that we generated in previous steps?

Reply:
yes, we can change that out. This was previously two different jobs, and so it needed a upload / download step to get the files again. Nice catch